src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.hpp

```diff
@@ -130,81 +141,123 @@ class FlightTaskAutoFollowTarget : public FlightTask
-	float _follow_angle_rad{0.0f};
-	AlphaFilter<float> _follow_angle_filter;
+	// Sticks object to read in stick commands from the user
+	Sticks _sticks;
 
 	// Estimator for target position and velocity
+	TargetEstimator _target_estimator;
 	follow_target_estimator_s _follow_target_estimator;
-	matrix::Vector2f _target_velocity_unit_vector;
 
-	// Lowpass filters for smoothingtarget position because it's used for setpoint generation
-	AlphaFilter<matrix::Vector3f> _target_position_filter;
+	// Last target estimator timestamp to handle timeout filter reset
+	uint64_t _last_valid_target_estimator_timestamp{0};
+
+	// Second Order Filter to calculate kinematically feasible target position
+	SecondOrderReferenceModel<matrix::Vector3f> _target_pose_filter;
+
+	// Internally tracked Follow Target characteristics, to allow RC control input adjustments
+	float _follow_target_distance{0.0f};
+	float _follow_target_height{0.0f};
+	float _follow_angle_rad{0.0f}; // 0 degrees following from front, and then clockwise rotation
+
+	// Estimated (Filtered) target orientation setpoint
+	float _target_orientation_rad{0.0f};
+
+	// Orbit angle setpoint, measured in global frame, against the target
+	float _orbit_angle_setpoint{0.0f};
+
+	// Actual drone to target heading [rad]
+	float _drone_to_target_heading{0.0f};
+
+	// Actual orbit angle in relation to the target
+	float _measured_orbit_angle{0.0f};
 
-	// Lowpass filter for smoothing the offset vector and have more dynamic shots when target changes direction
-	AlphaFilter<matrix::Vector2f> _offset_vector_filter;
+	// Actual drone to target 2d position vector
+	Vector2f _drone_to_target_vector{0.0f, 0.0f};
 
-	// Lowpass filter assuming  values 0-1, for avoiding big steps in velocity feedforward
-	AlphaFilter<float> _velocity_ff_scale;
+	// Tracked orbit tangential speed, to compensate for the orbital motion for velocity setpoints
+	Vector2f _orbit_tangential_velocity{0.0f, 0.0f};
 
 	// NOTE: If more of these internal state variables come into existence, it
 	// would make sense to create an internal state machine with a single enum
 	bool _emergency_ascent = false;
 
-	DEFINE_PARAMETERS(
-		(ParamFloat<px4::params::NAV_MIN_FT_HT>) _param_nav_min_ft_ht,
+	// Yaw setpoint filter to remove jitter-ness
+	AlphaFilter<float> _yaw_setpoint_filter;
+
+	// Variable to remember the home position's z coordinate, which will be baseline for the position z setpoint
+	float _home_position_z{0.0f};
+
+	float _gimbal_pitch{0};
+
+	DEFINE_PARAMETERS_CUSTOM_PARENT(
+		FlightTask,
+		(ParamFloat<px4::params::NAV_FT_HT>) _param_nav_ft_ht,
 		(ParamFloat<px4::params::NAV_FT_DST>) _param_nav_ft_dst,
 		(ParamInt<px4::params::NAV_FT_FS>) _param_nav_ft_fs,
 		(ParamInt<px4::params::NAV_FT_ALT_M>) _param_nav_ft_alt_m,
-		(ParamInt<px4::params::NAV_FT_GMB_M>) _param_nav_ft_gmb_m,
-		(ParamInt<px4::params::NAV_FT_DELC>) _param_nav_ft_delc
+		(ParamFloat<px4::params::NAV_FT_YAW_T>) _param_ft_yaw_t
```

Comment:
since these are all defined locally.. would be great to give the local states more verbose names so that we can actually understand what they are within this flight task. e.g. `_param_follow_target_height` (only with param prefix if the idea is that we want it to be clear this was from the params.. and we have other sources that could change the actually used follow target height)

Reply:
i would get rid of all acronyms, so that the variable is readable in the code without needing to go check the para.c file -- the param import here already tells where the value comes from, now in the flight task we just need something we can understand in the lines of the code. the only reason i would put the prefix param is to distinguish between "param" set value and the actual follow height that we are commanding (which can be adjusted from sticks as well). so there would in principle be two variables of the same name, unless we distinguish between the two.

Reply:
There was this discussion 3 years ago exactly about this issue. And it removed all these 'readable' parameter variable names in the code, and reset them to match exactly the parameter name. Could you review this as well for the final decision? thinking @tstastny @potaito

https://github.com/PX4/PX4-Autopilot/pull/11686

Reply:
Right I agree that using the direct parameter name in the code decreases code readability :yum:

Reply:
I have to admit I'm siding with #11686 because it's weird to sometimes be able to CTRL+F certain parameters, and other times not, because the handle is not re-using the parameter name :sweat_smile: 

But your argument is valid.

Reply:
@tstastny Thanks for asking. I totally understand your point and the verbose internal variable names were used for a very long time before it was streamlined. But for years I was wasting minutes every day looking up which parameter is used where and what exact parameter contributes to the code you're reading. I have to say it was such a big improvement when the names were streamlined and finally grepable, I never looked back once. As you can see in the pr this relief was shared by a lot of if not all the core contributors that cared. Correct me if I'm wrong. So I would disadvise reverting this naming strategy.

That said I don't like abbreviations in variable names and I encourage to use descriptive unshortened names. Also, I suggest thinking twice when defining or reviewing parameter names to make them as descriptive as possible given the limitations. Ideally parameter names would not be limited to 16 characters only.

Reply:
thanks for the extra insight @MaEtUgR and @potaito  - i can understand the reasoning. i guess then as we discussed in the dev call we should attack the *heart of the issue .. 16 char limits on param names. I'll follow up with @bkueng offline. @junwoo091400 for now you should then stick with the convention from https://github.com/PX4/PX4-Autopilot/pull/11686